jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java

```diff
@@ -197,36 +199,55 @@ static int getVersionFromString(String versionString) {
+      RegistryImage baseImage, AbsoluteUnixPath appRoot, ContainerizingMode containerizingMode)
+      throws IOException {
+    JavaContainerBuilder javaContainerBuilder =
+        JavaContainerBuilder.from(baseImage).setAppRoot(appRoot);
+
     try {
       if (isWarProject()) {
         Path explodedWarPath =
-            Paths.get(project.getBuild().getDirectory()).resolve(project.getBuild().getFinalName());
-        return JavaContainerBuilderHelper.fromExplodedWar(baseImage, explodedWarPath, appRoot);
+            Paths.get(project.getBuild().getDirectory(), project.getBuild().getFinalName());
+        return JavaContainerBuilderHelper.fromExplodedWar(javaContainerBuilder, explodedWarPath);
+      }
+
+      // Add dependencies
+      Set<Artifact> artifacts = project.getArtifacts();
+      List<Path> dependencies =
+          artifacts.stream().map(Artifact::getFile).map(File::toPath).collect(Collectors.toList());
+      javaContainerBuilder.addDependencies(dependencies);
+
+      switch (containerizingMode) {
+        case EXPLODED:
+          // Add resources, and classes
+          Path classesOutputDirectory = Paths.get(project.getBuild().getOutputDirectory());
+          // Don't use Path.endsWith(), since Path works on path elements.
+          Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+          javaContainerBuilder
+              .addResources(classesOutputDirectory, isClassFile.negate())
+              .addClasses(classesOutputDirectory, isClassFile);
+          break;
+
+        case PACKAGED:
+          // Add a JAR
+          javaContainerBuilder.addToClasspath(getJarArtifact());
+          break;
+
+        default:
+          throw new IllegalStateException("unknown containerizing mode: " + containerizingMode);
       }
 
-      Path classesOutputDirectory = Paths.get(project.getBuild().getOutputDirectory());
-      Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
-
-      // Add dependencies, resources, and classes
-      return JavaContainerBuilder.from(baseImage)
-          .setAppRoot(appRoot)
-          .addResources(classesOutputDirectory, isClassFile.negate())
-          .addClasses(classesOutputDirectory, isClassFile)
-          .addDependencies(
-              project
-                  .getArtifacts()
-                  .stream()
-                  .map(Artifact::getFile)
-                  .map(File::toPath)
-                  .collect(Collectors.toList()))
-          .toContainerBuilder();
+      return javaContainerBuilder.toContainerBuilder();
 
     } catch (IOException ex) {
       throw new IOException(
-          "Obtaining project build output files failed; make sure you have compiled your project "
+          "Obtaining project build output files failed; make sure you have "
+              + (containerizingMode == ContainerizingMode.PACKAGED ? "packaged" : "compiled")
```

Comment:
Is it worth pulling these out as a variable (e.g., `String necessaryGoal = ...`)

Reply:
Sadly, one is a past-tense verb (ends with "-ed"), so they are slightly different.